src/server/pkg/tarutil/util.go

```diff
@@ -0,0 +1,251 @@
+package tarutil
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+	"github.com/pachyderm/pachyderm/src/server/pkg/tar"
+)
+
+type File interface {
+	Header() (*tar.Header, error)
+	Content(io.Writer) error
+}
+
+type file struct {
+	hdr  *tar.Header
+	data []byte
+}
+
+func NewFile(name string, data []byte) File {
+	return newFile(NewHeader(name, int64(len(data))), data)
+}
+
+func newFile(hdr *tar.Header, data []byte) File {
+	return &file{
+		hdr:  hdr,
+		data: data,
+	}
+}
+
+func NewHeader(name string, size int64) *tar.Header {
+	return &tar.Header{
+		Name: name,
+		Size: size,
+	}
+}
+
+func (f *file) Header() (*tar.Header, error) {
+	return f.hdr, nil
+}
+
+func (f *file) Content(w io.Writer) error {
+	_, err := w.Write(f.data)
+	return err
+}
+
+type streamFile struct {
+	hdr *tar.Header
+	r   io.Reader
+}
+
+func NewStreamFile(name string, size int64, r io.Reader) File {
```

Comment:
`NewFileFromReader` would probably be more discoverable.

Reply:
I would prefer to keep this consistent with the other constructor.